docs/conf.py

```diff
@@ -161,6 +168,33 @@
 # For example: "explanation/old-name.html": "../how-to/prettify.html",
 # redirects = {}
 
+extensions.extend(
+    (
+        "sphinx.ext.ifconfig",
+        "sphinxcontrib.details.directive",
+    )
+)
+
+exclude_patterns.extend(
+    (
+        # Excluded because Snapcraft doesn't use overlays
+        "common/craft-parts/overlay_parameters.rst",
+        # Excluded here because they are included explicitly in other documents;
+        # if we don't exclude them, they generate "duplicate label" errors.
+        "common/craft-parts/dump_plugin.rst",
+        "common/craft-parts/part_properties.rst",
+        "common/craft-parts/python_plugin.rst",
+        "common/craft-parts/rust_plugin.rst",
+        "common/craft-parts/step_execution_environment.rst",
+        "common/craft-parts/step_output_directories.rst",
+        "common/craft-parts/explanation/filesets.rst",
+        "common/craft-parts/explanation/lifecycle.rst",
+        "common/craft-parts/explanation/parts.rst",
+        "common/craft-parts/explanation/how_parts_are_built.rst",
+        "common/craft-parts/reference/parts_steps.rst",
```

Comment:
reminder to add a comment boundary here or something for the next person adding an excludes